Meshtastic/Extensions/CoreData/MyInfoEntityExtension.swift

```diff
@@ -5,9 +5,35 @@
 //  Copyright(c) Garth Vander Houwen 9/3/23.
 //
 
-import Foundation
+import CoreData
+import MeshtasticProtobufs
 
 extension MyInfoEntity {
+	convenience init(
+		context: NSManagedObjectContext,
+		myInfo: MyNodeInfo,
+		peripheralId: String
+	) {
+		self.init(context: context)
+		self.myNodeNum = Int64(myInfo.myNodeNum)
+		self.rebootCount = Int32(myInfo.rebootCount)
+		self.minAppVersion = Int32(myInfo.minAppVersion)
+		self.peripheralId = peripheralId
+	}
+	
+	var adminIndex: Int32 {
+		guard let channels else { return 0 }
+		for (index, channel) in channels.enumerated() {
+			if (channel as? ChannelEntity)?.name?.lowercased() == "admin" {
+				return Int32(index)
+			}
+		}
+		return 0
+	}
+	
+	var bleName: String? {
+		myInfoNode?.user?.longName
+	}
```

Comment:
I was able to move these to computed properties, so we don't have to rely on saving them into this particular entity. Should simplify the data model a bit